src/PrestaShopBundle/Form/Admin/Extension/ModifyAllShopsExtension.php

```diff
@@ -0,0 +1,160 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Extension;
+
+use PrestaShop\PrestaShop\Core\Multistore\MultistoreContextCheckerInterface;
+use PrestaShopBundle\Form\FormBuilderModifier;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * This extension adds a new modify_all_shops option for form fields, then when the
+ * form is built a checkbox is automatically added which matches the configured field.
+ * This checkbox can then be used to apply the modification from the field to all shops.
+ */
+class ModifyAllShopsExtension extends AbstractTypeExtension
+{
+    public const MODIFY_ALL_SHOPS_PREFIX = 'modify_all_shops_';
+
+    /**
+     * @var MultistoreContextCheckerInterface
+     */
+    private $multistoreContextChecker;
+
+    /**
+     * @var FormBuilderModifier
+     */
+    private $formBuilderModifier;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    private $checkboxLabel;
+
+    public function __construct(
+        MultistoreContextCheckerInterface $multistoreContextChecker,
+        FormBuilderModifier $formBuilderModifier,
+        TranslatorInterface $translator
+    ) {
+        $this->multistoreContextChecker = $multistoreContextChecker;
+        $this->formBuilderModifier = $formBuilderModifier;
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getExtendedTypes(): iterable
+    {
+        return [
+            // To add the option on all form types
+            FormType::class,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if (!$this->multistoreContextChecker->isSingleShopContext()) {
+            return;
+        }
+
+        $isOverridden = $builder->getOption('modify_all_shops');
+        if ($isOverridden) {
+            $label = $this->getCheckboxLabel();
+            $builder->addEventListener(FormEvents::PRE_SET_DATA, static function (FormEvent $event) use ($label) {
```

Comment:
Why static?

Reply:
No reason! I just copied it from `MultistoreConfigurationTypeExtension` and I didn't notice I fixed both code